lib/@empirica/core/src/admin/classic/models.ts

```diff
@@ -684,6 +684,44 @@ export class Player extends GameOwned {
     return this.scopeByKey<PlayerStage>(key);
   }
 
+  get connectivityInfo() {
+    const key = `playerConnectivity-${this.id}`;
+
+    return this.scopeByKey<PlayerConnectivityInfo>(key);
+  }
+
+  async updateConnectivityInfo({ isConnected, connectedAt, disconnectedAt}: { isConnected: boolean, connectedAt: string | null, disconnectedAt: string | null} ) {
+    const key = `playerConnectivity-${this.id}`;
```

Comment:
I would not add methods on Player to update status, because now these method are available publicly.